src/typing/env_js.ml

```diff
@@ -75,33 +80,58 @@ let find_env ?(for_type=false) cx x reason =
   let rec loop = function
     | [] -> find_global cx x reason
     | block::blocks ->
-        (match SMap.get x !block with
+        let entries = block.entries in
+        (match SMap.get x !entries with
         | Some entry when (not entry.for_type || for_type) -> block
         | _ -> loop blocks)
   in
   loop (!env)
 
 let get_block x block =
```

Comment:
Probably time to rename this to `get_from_scope`
(not your doing, but while you're in here let's clean things up for clarity a bit)

Maybe these terminology renames would make more sense as a precursory pull request to simplify the real work going on in this one? As mentioned above, though, I think they become even more necessary now that 'block' has quite a different meaning after this work.